src/main/java/bio/terra/service/dataset/flight/LockDatasetStep.java

```diff
@@ -41,9 +52,14 @@ public LockDatasetStep(DatasetDao datasetDao, UUID datasetId,
 
     @Override
     public StepResult doStep(FlightContext context) {
+        DataAccessException faultInsert = getFaultToInsert();
         try {
             if (sharedLock) {
-                datasetDao.lockShared(datasetId, context.getFlightId());
+                try {
+                    datasetDao.lockShared(datasetId, context.getFlightId(), faultInsert);
```

Comment:
Rather than passing in an exception to the DatasetDao, could you autowire the ConfigurationService into the DatasetDao constructor and put the testInsertFault code in that class instead? I think it's a little cleaner to have the fault code closer to where the fault is exercised.

Reply:
Much cleaner, thank you!
I think this especially makes sense since this is the only place in our non-test code that hits lockShared. I think I was originally trying to avoid adding that logic to the root method to avoid unneeded computations.